pkg/rpc/context.go

```diff
@@ -910,7 +912,7 @@ func (d *delayingConn) Read(b []byte) (n int, err error) {
 		}
 		// If we somehow don't get our expected magic, throw an error.
 		if hdr.Magic != magic {
-			panic(errors.New("didn't get expected magic bytes header"))
+			return 0, errors.WithStack(errMagicNotFound)
```

Comment:
any reason not to use `errors.New("didn't get expected magic bytes header")` directly here?

Reply:
`errors.New` does more work